packages/core/tests/repository/data-source/in-memory.data-source.test.ts

```diff
@@ -1,87 +1,238 @@
-import {InMemoryDataSource, KeyQuery, NotFoundError, PaginationPageQuery, QueryNotSupportedError} from '../../../src';
+import {AllObjectsQuery, IdQuery, IdsQuery, InMemoryDataSource, InvalidArgumentError, KeyQuery, NotFoundError, PaginationPageQuery, QueryNotSupportedError, VoidQuery} from '../../../src';
 
 class SimplestClass {
     constructor(public readonly id: number) {
     }
 }
 
+class SimplestQuery extends KeyQuery {
+    constructor(value: SimplestClass) {
+        super(value.id.toString());
+    }
+}
+
```

Comment:
Why not just use the `IdQuery`, instead of creating a specific `Query` for the tests?